presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PickTableLayout.java

```diff
@@ -212,7 +208,26 @@ public Result apply(TableScanNode tableScanNode, Captures captures, Context cont
                 return Result.empty();
             }
 
-            return Result.ofPlanNode(planTableScan(tableScanNode, TRUE_LITERAL, context.getSession(), context.getSymbolAllocator().getTypes(), context.getIdAllocator(), metadata, parser, domainTranslator));
+            Optional<TableLayoutResult> layout = metadata.getLayout(
```

Comment:
@highker : I guess @martint  did this (in https://github.com/prestosql/presto/pull/363/commits/1285a778d2626e098cbd78b9d0ded28e43f41cbc) because the optimization rule name is `PickTableLayoutWithoutPredicate` :)